graphxblock/graphxblock.py

```diff
@@ -36,28 +102,61 @@ def student_view(self, context=None):
         The primary view of the GraphXBlock, shown to students
         when viewing courses.
         """
-        desmos_cdn_url = f"https://www.desmos.com/api/v1.9/calculator.js?apiKey={self.api_key}"
+        desmos_api = configuration_helpers.get_value("DESMOS_API_KEY", None)
+        if self.api_key:
+            desmos_api = self.api_key
+        desmos_cdn_url = (
+            f"https://www.desmos.com/api/v1.9/calculator.js?apiKey={desmos_api}"
```

Comment:
@farhaanbukhsh Could you read this url also from the site config with the current value being the default if nothing is passed? That should allow the option of using self-hosted APIs as well.